src/main/java/hudson/tasks/junit/TestResultAction.java

```diff
@@ -86,8 +88,14 @@ public TestResultAction(TestResult result, BuildListener listener) {
         this((Run) null, result, listener);
     }
 
+    @SuppressWarnings("deprecation")
     @Override public Collection<? extends Action> getProjectActions() {
-		return Collections.<Action>singleton(new TestResultProjectAction(run.getParent()));
+        Job<?,?> job = run.getParent();
+        if (/* getAction(Class) produces a StackOverflowError */!Util.filter(job.getActions(), TestResultProjectAction.class).isEmpty()) {
```

Comment:
If it comes from `JobProperty.getJobActions` (as in `XUnitPublisher`), or is added by `TransientProjectActionFactory`, it will be in `Job.getActions()` and thus we will not add a second copy—that is the purpose of this PR.